src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/SigningPropertyPage/AssemblyOriginatorKeyFileValueProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    [ExportInterceptingPropertyValueProvider]
+    internal sealed class AssemblyOriginatorKeyFileValueProvider : InterceptingPropertyValueProviderBase
+    {
+        public override string GetPropertyName() => "AssemblyOriginatorKeyFile";
+
+        public override Task<string> OnSetPropertyValueAsync(string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string> dimensionalConditions = null)
+        {
+            try
+            {
+                var projectFileFullPath = defaultProperties.FileFullPath;
+                if (!string.IsNullOrEmpty(projectFileFullPath) && PathUtilities.IsAbsolute(unevaluatedPropertyValue))
```

Comment:
Instead of copying Roslyn's code here, can we just use existing proejct system code, in particular, PathHelper.TryMakeRelativeToProjectDirectory and PathHelper.XXXRooted API?